### PR TITLE
Allow support for architectures other than x86_64

### PR DESCRIPTION
**Pull request by nkrinner**

### CODE REVIEW

**Comment by vuntz** [comment]
+1. Tested with https://github.com/crowbar/crowbar-core/pull/83 and https://github.com/SUSE-Cloud/automation/pull/613

**Comment by abel-navarro** [comment]
there are several updates to point to /srv/tftpboot/.../$arch. Won't this break compatibility with cloud6, m{1..6} deployments?

**Comment by vuntz** [comment]
> there are several updates to point to /srv/tftpboot/.../$arch. Won't this break compatibility with cloud6, m{1..6} deployments?

People upgrading from an earlier milestone will have to change a few things in the directories layout of their system, yes. It's acceptable because we're still in the milestones phase, would be bad if that was after the release.

**Comment by vuntz** [comment]
I've rebased that on top of https://github.com/crowbar/crowbar/pull/2172 at https://github.com/MaximilianMeister/crowbar/compare/activate-repos...vuntz:multi-arch-yaml?expand=1

**Comment by vuntz** [comment]
@nkrinner can you rebase? Or maybe just take the commits from my branch (see previous comment)

**Comment by nkrinner** [comment]
Rebased

File: scripts/repos-check-suse [hunk 1]
```diff
@@ -90,6 +90,71 @@ check_media_content () {
...
+                echo "Ignoring failed repo check for $repo ($version - $arch) due to \$REPOS_SKIP_CHECKS ($xml is missing $tag tag '$expected')"
+                return 0
+            fi
+            die "$repo ($version - $arch) does not contain the right repository ($xml is missing $tag tag '$expected')"
+        fi
+    fi
+
+    check_repo_key $version $arch $repo $ignore_failure
+}
+
```
Comment (vuntz) [inline scripts/repos-check-suse]: That's wrong, you should not add that back.

File: scripts/install-chef-suse.sh [hunk 2]
```diff
@@ -491,81 +493,118 @@ if [ -z "$(type -t skip_check_for_repo)" ]; then
...
-    MEDIA=/srv/tftpboot/suse-12.1/install
+        check_repo_tag repo    12.1 $arch SLES12-SP1-Pool                     "obsproduct://build.suse.de/SUSE:SLE-12-SP1:GA/SLES/12.1/POOL/$arch"
+        check_repo_tag repo    12.1 $arch SLES12-SP1-Updates                  "obsrepository://build.suse.de/SUSE:Updates:SLE-SERVER:12-SP1:$arch/update"
+        check_repo_tag repo    12.1 $arch SUSE-OpenStack-Cloud-6-Pool         "obsproduct://build.suse.de/SUSE:SLE-12:SP1:Products:Cloud6/suse-openstack-cloud/6/POOL/$arch" $REQUIRE_CLOUD
+        check_repo_tag repo    12.1 $arch SUSE-OpenStack-Cloud-6-Updates      "obsrepository://build.suse.de/SUSE:Updates:OpenStack-Cloud:6:$arch/update" $REQUIRE_CLOUD
+        # HA is s390x/x86_64 only
+        if [ $arch == x86_64 -o $arch == s390x ]; then
+            check_repo_tag repo    12.1 $arch SLE12-SP1-HA-Pool                   "obsproduct://build.suse.de/SUSE:SLE-12-SP1:GA/sle-ha/12.1/POOL/$arch" 'false'
+            check_repo_tag repo    12.1 $arch SLE12-SP1-HA-Updates                "obsrepository://build.suse.de/SUSE:Updates:SLE-HA:12-SP1:$arch/update" 'false'
+        fi
```
Comment (vuntz) [inline scripts/install-chef-suse.sh]: There should be no check_repo_tag anymore...

Comment (vuntz) [inline scripts/install-chef-suse.sh]: See https://github.com/vuntz/crowbar/tree/multi-arch-yaml for how it should be

Comment (vuntz) [inline scripts/install-chef-suse.sh]: Actually, see https://github.com/nkrinner/crowbar/pull/5 :-)

**Comment by nkrinner** [comment]
Aparently I got something wrong here. Thanks for @vuntz for providing a fix.

**Comment by vuntz** [comment]
+1

**Comment by jsuchome** [comment]
+1. Tested with crowbar/crowbar-core#83 and SUSE-Cloud/automation#613

**Pull request merged**